versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractReferences.java

```diff
@@ -69,12 +69,13 @@ public void createAndDeleteBranch() throws Exception {
     soft.assertThat(hash).isNotNull();
 
     final BranchName anotherBranch = BranchName.of("bar");
-    final Hash createHash = store().create(anotherBranch, Optional.of(hash));
+    final Hash createHash = store().create(anotherBranch, Optional.of(hash)).getHash();
```

Comment:
Can you add some assertion on the `ReferenceCreatedResult` here and on `ReferenceDeletedResult` to the (unasserted) `.delete()` in line 103? And similar for tags below?